docs/src/components/DocNavigation/index.tsx

```diff
@@ -0,0 +1,190 @@
+import React from "react";
+import { useStaticQuery, graphql } from "gatsby";
+
+import Wrapper, { WrapperDesktop, WrapperMobile } from "./primitives/Wrapper";
+import Collapse from "./primitives/Collapse";
+import Category from "./primitives/Category";
+import Link from "./primitives/Link";
+import useDevMode from "../../hooks/useDevMode";
+
+type Trail = Array<{
+  name: string;
+  url: string;
+}>;
+
+interface QueryData {
+  allMdx: {
+    nodes: Array<{
+      id: string;
+      fields: {
+        tabCollection: string | null;
+        trail: Trail;
+      };
+      parent: {
+        name: string;
+      };
+    }>;
+  };
+}
+
+type NavigationItem =
+  | {
+      type: "branch";
+      name: string;
+      items: Navigation;
+    }
+  | {
+      type: "leaf";
+      name: string;
+      url: string;
+    };
+
+type Navigation = NavigationItem[];
+
+export function groupTrails(trails: Trail[]): Navigation {
+  // https://stackoverflow.com/a/57344801/1247274
+  const result: Navigation = [];
+  const level = { result };
+  trails.forEach(trail => {
+    trail.reduce((acc, { name, url }, index) => {
+      if (!acc[name]) {
+        acc[name] = { result: [] };
+        if (index < trail.length - 1) {
+          acc.result.push({ type: "branch", name, items: acc[name].result });
+        } else {
+          acc.result.push({ type: "leaf", name, url });
+        }
+      }
+      return acc[name];
+    }, level);
+  });
+  return result;
+}
+
+function getItemKey(item: NavigationItem) {
+  if (item.type === "branch") {
+    return `${item.type}-${item.name}`;
+  }
+  return item.url;
+}
+
+function DocNavigationItem({
+  devMode,
+  currentUrl,
+  level,
+  item,
+}: {
+  devMode: boolean;
+  currentUrl: string;
+  level: number;
+  item: NavigationItem;
+}) {
+  const [expanded, setExpanded] = React.useState(
+    level === 1 &&
+      item.type === "branch" &&
+      item.items.some(it =>
+        it.type === "leaf"
+          ? currentUrl.startsWith(it.url)
+          : it.items.some(i => i.type === "leaf" && currentUrl.startsWith(i.url)),
+      ),
```

Comment:
this looks a bit complicated for a default state 🤔  
it should be assigned to some variable at least